arangod/Cluster/ClusterInfo.cpp

```diff
@@ -786,8 +787,34 @@ void ClusterInfo::loadPlan() {
     name = database.first;
     auto dbSlice = database.second->slice()[0];
     std::vector<std::string> dbPath{AgencyCommHelper::path(), "Plan", "Databases", name};
-    if (!dbSlice.hasKey(dbPath)) { // Dropped from Plan
+
+    // Dropped from Plan?
+    if (!dbSlice.hasKey(dbPath)) {
+      std::shared_ptr<VPackBuilder> plan = nullptr;
+      {
+        READ_LOCKER(guard, _planProt.lock);
+        auto it = _plan.find(name);
+        if (it != _plan.end()) {
+          plan = it->second;
+        }
+      }
+      if (plan != nullptr) {
+        std::vector<std::string> colPath{AgencyCommHelper::path(), "Plan", "Collections", name};
+        if (plan->slice()[0].hasKey(colPath)) {
+          for (auto const& col : VPackObjectIterator(plan->slice()[0].get(colPath))) {
```

Comment:
This is inefficient, as the sequence `plan->slice()[0][colPath]` needs to be evaluated twice, and this can be slow in VelocyPack.
Modifying it as follows will make it much more efficient:
```suggestion
        auto planColSlice = plan->slice()[0].get(colPath);
        if (planColSlice.isObject()) {
          for (auto const& col : VPackObjectIterator(planColSlice)) {
```